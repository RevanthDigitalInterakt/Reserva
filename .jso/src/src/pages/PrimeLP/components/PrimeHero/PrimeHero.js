  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[1]);
  var _EventProvider = _$$_REQUIRE(_dependencyMap[2])(_$$_REQUIRE(_dependencyMap[3]));
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[2])(_$$_REQUIRE(_dependencyMap[4]));
  var _IconPrimeLogo = _$$_REQUIRE(_dependencyMap[2])(_$$_REQUIRE(_dependencyMap[5]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function PrimeHero(_ref) {
    var data = _ref.data,
      onAddToCart = _ref.onAddToCart;
    var _usePrimeInfo = (0, _$$_REQUIRE(_dependencyMap[6]).usePrimeInfo)(),
      isPrime = _usePrimeInfo.isPrime;
    var onPressAddCart = (0, _react.useCallback)(function () {
      _EventProvider.default.logEvent('prime_press_add_to_cart_lp', {
        position: 'top'
      });
      onAddToCart();
    }, [onAddToCart]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.ImageBackground, {
      source: _$$_REQUIRE(_dependencyMap[8]),
      style: {
        width: _configDeviceSizes.default.DEVICE_WIDTH
      },
      resizeMode: "cover",
      testID: "com.usereserva:id/PrimeHero_component",
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
        style: _$$_REQUIRE(_dependencyMap[9]).styles.wrapper,
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_IconPrimeLogo.default, {
          style: _$$_REQUIRE(_dependencyMap[9]).styles.icon
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[10]).Typography, {
          variant: "subtituloSessoes",
          style: _$$_REQUIRE(_dependencyMap[9]).styles.title,
          children: ["Um para\xEDso de benef\xEDcios s\xF3", '\n', `seus por ${data.installmentQty}x`, /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Typography, {
            variant: "descontoTag2",
            style: _$$_REQUIRE(_dependencyMap[9]).styles.bold,
            children: ` ${data.installmentPrice}/MÃªs*.`
          })]
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
          style: _$$_REQUIRE(_dependencyMap[9]).styles.buttonContainer,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.TouchableOpacity, {
            testID: "com.usereserva:id/PrimeHero_call_to_action",
            style: _$$_REQUIRE(_dependencyMap[9]).styles.button,
            onPress: onPressAddCart,
            children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Typography, {
              variant: "tituloSessao",
              style: _$$_REQUIRE(_dependencyMap[9]).styles.buttonText,
              children: isPrime ? 'CONTINUAR COMPRANDO' : 'ASSINE AGORA'
            })
          })
        })]
      })
    });
  }
  var _default = exports.default = PrimeHero;
