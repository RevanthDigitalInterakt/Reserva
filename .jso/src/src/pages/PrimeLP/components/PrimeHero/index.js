  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeHero.default;
    }
  });
  var _PrimeHero = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
