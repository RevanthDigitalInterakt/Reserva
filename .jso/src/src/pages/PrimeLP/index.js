  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeLP.default;
    }
  });
  var _PrimeLP = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
