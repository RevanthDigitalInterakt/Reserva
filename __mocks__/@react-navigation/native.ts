const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
