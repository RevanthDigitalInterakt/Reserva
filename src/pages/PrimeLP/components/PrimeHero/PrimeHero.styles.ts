import { StyleSheet } from 'react-native';
import { COLORS } from '../../../../base/styles/colors';

export const styles = StyleSheet.create({
  wrapper: {
    alignItems: 'center',
    paddingTop: 65,
    paddingBottom: 110,
    paddingHorizontal: 20,
  },

  icon: {
    marginBottom: 10,
  },

  title: {
    fontSize: 18,
    lineHeight: 23,
    textAlign: 'center',
    color: COLORS.WHITE,
    marginBottom: 24,
  },

  bold: {
    fontSize: 18,
    lineHeight: 23,
    fontWeight: '900',
  },

  subtitle: {
    marginBottom: 32,
    fontSize: 12.5,
    lineHeight: 18.5,
    textAlign: 'center',
    color: COLORS.WHITE,
    paddingHorizontal: 10,
  },

  buttonContainer: {
    paddingHorizontal: 20,
    width: '100%',
  },

  button: {
    backgroundColor: COLORS.WHITE,
    alignItems: 'center',
    justifyContent: 'center',
    height: 40,
  },

  buttonText: {
    fontSize: 12,
    lineHeight: 14,
    letterSpacing: 0.5,
    fontWeight: '700',
  },
});
