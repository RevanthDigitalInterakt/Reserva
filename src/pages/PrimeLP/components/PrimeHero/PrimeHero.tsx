import React, { useCallback } from 'react';
import { ImageBackground, TouchableOpacity, View } from 'react-native';
import { Typography } from '@usereservaapp/reserva-ui';
import configDeviceSizes from '../../../../utils/configDeviceSizes';
import { styles } from './PrimeHero.styles';
import IconPrimeLogo from '../Icons/IconPrimeLogo';
import type { PrimeDetailOutput } from '../../../../base/graphql/generated';
import EventProvider from '../../../../utils/EventProvider';
import { useAuthStore } from '../../../../zustand/useAuth/useAuthStore';
import { usePrimeStore } from '../../../../zustand/usePrimeStore/usePrimeStore';

const ImageSource = require('../../../../../assets/common/header.png');

interface IPrimeHero {
  data: PrimeDetailOutput;
  onAddToCart: () => Promise<void>;
}

function PrimeHero({ data, onAddToCart }: IPrimeHero) {
  const { profile } = useAuthStore(['profile']);
  const { hasPrimeSubscriptionInCart } = usePrimeStore(['hasPrimeSubscriptionInCart']);

  const onPressAddCart = useCallback(() => {
    EventProvider.logEvent('prime_press_add_to_cart_lp', { position: 'top' });
    onAddToCart();
  }, [onAddToCart]);

  const hasPrime = profile?.isPrime || hasPrimeSubscriptionInCart;

  return (
    <ImageBackground
      source={ImageSource}
      style={{ width: configDeviceSizes.DEVICE_WIDTH }}
      resizeMode="cover"
      testID="com.usereserva:id/PrimeHero_component"
    >
      <View style={styles.wrapper}>
        <IconPrimeLogo style={styles.icon} />

        <Typography variant="subtituloSessoes" style={styles.title}>
          Um paraíso de benefícios só
          {'\n'}
          {`seus por ${data.installmentQty}x`}
          <Typography variant="descontoTag2" style={styles.bold}>
            {` ${data.installmentPrice}/Mês*.`}
          </Typography>
        </Typography>

        <Typography variant="tituloSessao" style={styles.subtitle}>
          Torne-se membro do maior clube de vantagens da
          {'\n'}
          moda brasileira e tenha acesso a descontos
          especiais, frete grátis e muito mais.
        </Typography>

        <View style={styles.buttonContainer}>
          <TouchableOpacity
            testID="com.usereserva:id/PrimeHero_call_to_action"
            style={styles.button}
            onPress={onPressAddCart}
          >
            <Typography variant="tituloSessao" style={styles.buttonText}>
              {hasPrime ? 'CONTINUAR COMPRANDO' : 'ASSINE AGORA'}
            </Typography>
          </TouchableOpacity>
        </View>
      </View>
    </ImageBackground>
  );
}

export default PrimeHero;
