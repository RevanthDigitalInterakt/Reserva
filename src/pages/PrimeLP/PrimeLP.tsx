import type { StackScreenProps } from '@react-navigation/stack';
import React, {
  useCallback, useMemo, useState,
} from 'react';
import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { useLandingPagePrimeQuery } from '../../base/graphql/generated';
import { ModalBag } from '../../components/ModalBag/ModalBag';
import { useApolloFetchPolicyStore } from '../../zustand/useApolloFetchPolicyStore';
import testProps from '../../utils/testProps';
import { ModalWelcomePrime } from '../../components/ModalWelcomePrime';
import { usePrimeInfo } from '../../hooks/usePrimeInfo';
import { TopBarDefaultBackButton } from '../../modules/Menu/components/TopBarDefaultBackButton';
import type { RootStackParamList } from '../../routes/StackNavigator';
import { usePrimeStore } from '../../zustand/usePrimeStore/usePrimeStore';
import PrimeBenefits from './components/PrimeBenefits';
import PrimeFAQ from './components/PrimeFAQ/PrimeFAQ';
import PrimeHero from './components/PrimeHero';
import PrimeIntro from './components/PrimeIntro';
import PrimeSubscribe from './components/PrimeSubscribe';
import { Box } from '../../components/Box/Box';
import { ExceptionProvider } from '../../base/providers/ExceptionProvider';

type IPrimeLP = StackScreenProps<RootStackParamList, 'PrimeLP'>;

function PrimeLP({ navigation }: IPrimeLP) {
  const { onAddPrimeToCart, isPrime } = usePrimeInfo();
  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);
  const [loadingAddCartPrime, setLoadingAddCartPrime] = useState(false);

  const {
    animationBag,
    handleClickContinue,
    isVisibleModalWelcome,
    changeStateAnimationBag,
    changeStateIsVisibleModalWelcome,
  } = usePrimeStore([
    'animationBag',
    'handleClickContinue',
    'isVisibleModalWelcome',
    'changeStateAnimationBag',
    'changeStateIsVisibleModalWelcome',
  ]);

  const { data: rawData, loading } = useLandingPagePrimeQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('landingPagePrime'),
  });

  const data = useMemo(() => rawData?.landingPagePrime, [rawData?.landingPagePrime]);

  const handleOnModalHideSignIn = useCallback(() => {
    changeStateAnimationBag(true);
  }, [changeStateAnimationBag]);

  const handleOnModalHide = useCallback(() => {
    if (isPrime) {
      setTimeout(() => {
        changeStateIsVisibleModalWelcome(true);
      }, 500);
    }
  }, [changeStateIsVisibleModalWelcome, isPrime]);

  const onAddToCart = useCallback(async () => {
    try {
      if (!data || loadingAddCartPrime) return;

      if (isPrime) {
        navigation.popToTop();
        return;
      }

      setLoadingAddCartPrime(true);

      await onAddPrimeToCart(false);

      handleOnModalHideSignIn();
    } catch (e) {
      ExceptionProvider.captureException(e);
    } finally {
      setLoadingAddCartPrime(false);
    }
  }, [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);

  const onCloseModalWelcomePrime = useCallback(() => {
    handleClickContinue();

    setTimeout(() => {
      navigation.navigate('Home');
    }, 300);
  }, [handleClickContinue, navigation]);

  return (
    <SafeAreaView style={{ backgroundColor: '#fff' }}>
      <Box bg="white">
        <TopBarDefaultBackButton
          loading={loading || loadingAddCartPrime}
          navigateGoBack
        />

        <ModalBag
          isVisible={animationBag}
          onBackdropPress={() => changeStateAnimationBag(false)}
          onModalHide={handleOnModalHide}
        />

        <ModalWelcomePrime
          isVisible={isVisibleModalWelcome}
          onClose={onCloseModalWelcomePrime}
        />

        {!!(data && !loading) && (
          <ScrollView
            contentContainerStyle={{ paddingBottom: 100 }}
            {...testProps('PrimeLP_page')}
          >
            <PrimeHero data={data} onAddToCart={onAddToCart} />

            <PrimeIntro data={data} />

            <PrimeBenefits data={data} />

            <PrimeSubscribe data={data} onAddToCart={onAddToCart} />

            <PrimeFAQ data={data} />
          </ScrollView>
        )}
      </Box>
    </SafeAreaView>
  );
}

export default PrimeLP;
