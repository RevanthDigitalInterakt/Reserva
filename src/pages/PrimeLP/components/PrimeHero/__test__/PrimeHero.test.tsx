import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import { fireEvent, render, screen } from '@testing-library/react-native';
import PrimeHero from '../PrimeHero';
import { mockPrimeData, addToCartMock } from '../../../../../../__mocks__/PrimeLP.mock';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <PrimeHero data={mockPrimeData} onAddToCart={addToCartMock} />
  </ThemeProvider>
);

describe('PrimeHero', () => {
  beforeEach(() => {
    render(TestingComponent);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render without error and match to snapshot', () => {
    const primeHero = screen.getByTestId(
      'com.usereserva:id/PrimeHero_component',
    );

    expect(primeHero).toBeOnTheScreen();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call to action button to be on the screen', async () => {
    const callToAction = screen.getByTestId(
      'com.usereserva:id/PrimeHero_call_to_action',
    );

    expect(callToAction).toBeOnTheScreen();
  });

  it('should call addPrimeToCart function', () => {
    const callToAction = screen.getByTestId(
      'com.usereserva:id/PrimeHero_call_to_action',
    );

    fireEvent.press(callToAction);

    expect(addToCartMock).toHaveBeenCalledTimes(1);
  });
});
