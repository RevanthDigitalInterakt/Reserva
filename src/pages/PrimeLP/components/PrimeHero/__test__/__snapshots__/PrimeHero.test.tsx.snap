// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeHero should render without error and match to snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "width": 750,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "commons": {
          "GuideBermuda": [Circular],
          "GuideCalca": [Circular],
          "GuideCamiseta": [Circular],
          "GuideHoodie": [Circular],
          "GuideJaqueta": [Circular],
          "GuideMangaCurta": [Circular],
          "GuideMangaLonga": [Circular],
          "GuidePolo": [Circular],
          "GuideShort": [Circular],
          "GuideSueter": [Circular],
          "GuideSungaCueca": [Circular],
          "QrcodeBackground": [Circular],
          "americanExpress": [Circular],
          "arrowInstruction": [Circular],
          "arrowRight": [Circular],
          "bagEmpty": [Circular],
          "bannerCatalog": [Circular],
          "bannerLogin": [Circular],
          "bannerOffer": [Circular],
          "cardFront": [Circular],
          "cardVerso": [Circular],
          "chat": [Circular],
          "checkedBox": [Circular],
          "close": [Circular],
          "clothingCare1": [Circular],
          "clothingCareBleach": [Circular],
          "clothingCareDrying": [Circular],
          "clothingCarePass": [Circular],
          "clothingCareProfessionalWashing": [Circular],
          "clothingCareWash": [Circular],
          "config": [Circular],
          "copy": [Circular],
          "cupomModalBackground": [Circular],
          "discountFlag": [Circular],
          "dividerReserva": [Circular],
          "document": [Circular],
          "elo": [Circular],
          "emptyProductCatalog": [Circular],
          "expand": [Circular],
          "facebook": [Circular],
          "faq": [Circular],
          "filledHeart": [Circular],
          "foraDoAr": [Circular],
          "greenCheck": [Circular],
          "headerLogin": [Circular],
          "heart": [Circular],
          "help": [Circular],
          "homeMock1": [Circular],
          "homeMock2": [Circular],
          "homeMock3": [Circular],
          "homeMock4": [Circular],
          "imageCashback": [Circular],
          "imageNotFound": [Circular],
          "instagram": [Circular],
          "ios14": [Circular],
          "localReserva": [Circular],
          "logoQRCode": [Circular],
          "mastercard": [Circular],
          "noStoresFound": [Circular],
          "noWishList": [Circular],
          "pinPlace": [Circular],
          "pinYou": [Circular],
          "profile": [Circular],
          "purchaseConfirmation": [Circular],
          "raceImageBackground": [Circular],
          "reservaIcon": [Circular],
          "roulet": [Circular],
          "saleOff": [Circular],
          "search": [Circular],
          "searchNotFound": [Circular],
          "selectRectangle": [Circular],
          "share": [Circular],
          "shipping1": [Circular],
          "shipping2": [Circular],
          "shirt1": [Circular],
          "shirt2": [Circular],
          "shirt3": [Circular],
          "shirt4": [Circular],
          "signup1": [Circular],
          "signup2": [Circular],
          "signup3": [Circular],
          "signup4": [Circular],
          "signup5": [Circular],
          "topBannerMock1": [Circular],
          "twitter": [Circular],
          "uncheckedBox": [Circular],
          "unfilledHeart": [Circular],
          "update": [Circular],
          "visa": [Circular],
          "whatsapp": [Circular],
          "withoutInternet": [Circular],
          "zoomHand": [Circular],
        },
        "default": {
          "GuideBermuda": [Circular],
          "GuideCalca": [Circular],
          "GuideCamiseta": [Circular],
          "GuideHoodie": [Circular],
          "GuideJaqueta": [Circular],
          "GuideMangaCurta": [Circular],
          "GuideMangaLonga": [Circular],
          "GuidePolo": [Circular],
          "GuideShort": [Circular],
          "GuideSueter": [Circular],
          "GuideSungaCueca": [Circular],
          "QrcodeBackground": [Circular],
          "americanExpress": [Circular],
          "arrowInstruction": [Circular],
          "arrowRight": [Circular],
          "bagEmpty": [Circular],
          "bannerCatalog": [Circular],
          "bannerLogin": [Circular],
          "bannerOffer": [Circular],
          "cardFront": [Circular],
          "cardVerso": [Circular],
          "chat": [Circular],
          "checkedBox": [Circular],
          "chevronRight": [Function],
          "close": [Circular],
          "clothingCare1": [Circular],
          "clothingCareBleach": [Circular],
          "clothingCareDrying": [Circular],
          "clothingCarePass": [Circular],
          "clothingCareProfessionalWashing": [Circular],
          "clothingCareWash": [Circular],
          "config": [Circular],
          "copy": [Circular],
          "cupomModalBackground": [Circular],
          "discountFlag": [Circular],
          "dividerReserva": [Circular],
          "document": [Circular],
          "elo": [Circular],
          "emptyProductCatalog": [Circular],
          "expand": [Circular],
          "facebook": [Circular],
          "faq": [Circular],
          "filledHeart": [Circular],
          "foraDoAr": [Circular],
          "greenCheck": [Circular],
          "headerLogin": [Circular],
          "heart": [Circular],
          "help": [Circular],
          "homeMock1": [Circular],
          "homeMock2": [Circular],
          "homeMock3": [Circular],
          "homeMock4": [Circular],
          "imageCashback": [Circular],
          "imageNotFound": [Circular],
          "info": [Function],
          "instagram": [Circular],
          "ios14": [Circular],
          "localReserva": [Circular],
          "logoPrime": [Function],
          "logoQRCode": [Circular],
          "mastercard": [Circular],
          "noStoresFound": [Circular],
          "noWishList": [Circular],
          "pinPlace": [Circular],
          "pinYou": [Circular],
          "profile": [Circular],
          "purchaseConfirmation": [Circular],
          "raceImageBackground": [Circular],
          "reservaIcon": [Circular],
          "roulet": [Circular],
          "saleOff": [Circular],
          "search": [Circular],
          "searchNotFound": [Circular],
          "selectRectangle": [Circular],
          "share": [Circular],
          "shipping1": [Circular],
          "shipping2": [Circular],
          "shirt1": [Circular],
          "shirt2": [Circular],
          "shirt3": [Circular],
          "shirt4": [Circular],
          "signup1": [Circular],
          "signup2": [Circular],
          "signup3": [Circular],
          "signup4": [Circular],
          "signup5": [Circular],
          "topBannerMock1": [Circular],
          "trash": [Function],
          "twitter": [Circular],
          "uncheckedBox": [Circular],
          "unfilledHeart": [Circular],
          "update": [Circular],
          "visa": [Circular],
          "whatsapp": [Circular],
          "withoutInternet": [Circular],
          "zoomHand": [Circular],
        },
        "svgs": {
          "chevronRight": [Function],
          "info": [Function],
          "logoPrime": [Function],
          "trash": [Function],
        },
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": 750,
        },
        undefined,
      ]
    }
    testID="com.usereserva:id/PrimeHero_component"
  />
  <View
    style={
      {
        "alignItems": "center",
        "paddingBottom": 110,
        "paddingHorizontal": 20,
        "paddingTop": 175,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="61"
      bbWidth="280"
      fill="none"
      focusable={false}
      height={61}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginBottom": 10,
          },
          {
            "flex": 0,
            "height": 61,
            "width": 280,
          },
        ]
      }
      width={280}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          clipPath="iconprimelogo"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M24.153 44.047v9.082l7.473 6.509H0l7.095-6.509V16.196L0 9.233h24.078c16.378 0 26.795 6.434 26.795 17.483 0 11.201-9.209 17.558-23.323 17.558-1.133 0-2.265-.075-3.397-.227Zm0-30.878v27.245c5.51-.605 9.058-4.313 9.058-12.866 0-9.46-2.491-13.32-9.058-14.379ZM89.428 26.338c0 5.373-2.717 7.795-5.887 7.795-3.397 0-4.605-2.725-7.473-2.725-1.962 0-3.095 1.438-3.095 4.465V53.28l6.87 6.358H51.16l6.19-6.358V32.317l-5.661-4.693 19.85-9.309c.453 4.768.755 9.158.83 13.85 1.208-11.504 5.737-13.85 9.813-13.85 4.453 0 7.246 2.8 7.246 8.023ZM96.183 53.13V31.332l-5.963-4.768 21.663-8.931V53.28l7.17 6.358H89.088l7.095-6.509Zm-.68-45.335c0-4.313 3.17-7.795 8.605-7.795 5.435 0 8.605 3.482 8.605 7.795 0 4.314-3.17 7.796-8.605 7.796-5.434 0-8.604-3.482-8.604-7.796ZM177.685 53.962v-18.24c0-4.238-1.66-5.903-5.888-5.903-1.735 0-3.245.227-4.452.606.074.605.074 1.21.074 1.892v20.812l5.134 6.509h-25.437l4.303-5.676v-18.24c0-4.238-1.586-5.903-5.888-5.903-1.736 0-3.171.227-4.378.53v23.613l4.454 5.676h-26.419l6.265-6.509V32.392l-5.208-4.768 20.908-9.309v9.99c3.698-6.887 8.001-9.914 13.737-9.914 6.793 0 11.247 3.406 12.303 10.369 3.774-7.19 8.152-10.37 14.266-10.37 7.472 0 12.227 4.542 12.227 13.927v20.812l5.058 6.509h-25.437l4.378-5.676ZM237.321 45.107C235.887 55.778 227.358 61 218.979 61c-10.038 0-21.285-7.568-21.285-21.115 0-12.412 9.36-21.04 19.625-21.04 10.718 0 17.662 6.736 18.115 15.288l-22.266 7.72c3.169 3.935 7.925 6.886 13.662 6.886 4.226 0 8-1.437 10.491-3.632Zm-28.154-13.774c0 2.421.68 4.92 1.887 7.34l10.341-5.146c-1.887-5.297-5.359-7.946-8.076-7.946-2.341 0-4.152 1.892-4.152 5.752Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M278.815 57.66c-.752-.593-4.174-2.858-5.839-4.346-.649-.58-.191-1.248-.907-1.944-.913-.89-1.926-1.853-2.925-2.881-1.05-1.08-2.034-2.274-2.988-3.511-.868-1.213-3.577-4.96-5.781-7.442-2.364-2.663-6.831-5.921-8.057-8.914a8.742 8.742 0 0 1-.201-.609c-.273-.958-.407-2.035-.368-3.279.039-1.208.396-2.431 1.731-4.468 1.456-2.221 1.195-3.199.689-3.277-.495-.077-1.035-.054-.494-2.374.467-2.005.141-2.844-.161-3.346-.285-.472-.636-.72-.91-.662-.63.133-1.105 1.61-1.72 2-.158.1-.963.553-.804-1.323.038-.443-.265-.596-.688-.333-.631.393-1.527 1.413-2.013 3.33-.031.119-.147.155-.201-.002-.08-.233-.111-.678.07-1.43.167-.7.454-1.261.135-1.338-.404-.098-.978.749-1.42 1.711a5.463 5.463 0 0 0-.134.308c-.048.12-.184.19-.194-.055-.007-.17.032-.441.124-.766.04-.142-.036-.2-.092-.215-.132-.034-.296.037-.498.188-.455.337-.896.956-1.213 1.776l-.003-.003c-.033.09-.063.182-.092.272l-.021.06-.027.087c-.015.047-.027.096-.041.144-.017.063-.037.126-.052.188a.933.933 0 0 0-.017.072l-.001.002a7.818 7.818 0 0 0-.154.848 7.346 7.346 0 0 0-.054 1.117v.003c.002.061.002.123.006.183.011.198-.468.252-.364.7.263 1.132.677 2.283.902 2.875a.4.4 0 0 1-.26.527c-1.054.31-3.862 1.197-5.837 2.395-.115.07-.204.262-.101.415.096.143.328.125.436.089 2.261-.752 5.484-1.81 7.041-.681 1.142.827 1.258 2.801.542 4.934-.237.791-.507 1.497-.754 2.22l-.01.004c-3.312 8.774 2.792 12.817 8.408 18.03.413.383 1.267 1.347 2.029 2.47.55.81 1.109 1.712 1.052 2.596-.047.744-.165 1.639-.423 2.796-.315 1.416-.656.718-2.884.153-.698-.177-.73-.113-1.173.836-.46.984-.498 1.096-.375 1.16.099.052 4.274.024 4.74.112a.86.86 0 0 1 .21.068c.507.24 2.804 1.342 3.504 1.812.15.1.38.064.468-.063.104-.152.026-.346-.108-.429-1.033-.64-3.122-2.222-3.175-2.707-.021-.186.147-.898.23-1.44.376-2.435.435-3.35 1.429-3.814.921-.432 3.075.672 4.768 1.086a.143.143 0 0 1 .107.17c-.054.241-.192.813-.478 1.785-.039.132-.142.444-.421.529-.359.108-1.778-.42-2.729-.648-.4-.096-.517-.356-1.422 1.553-.064.135-.04.186.003.22.055.043 4.03.24 4.594.35.102.02.201.052.295.098a80.67 80.67 0 0 1 3.497 1.83c.433.237.574.344.732.186.204-.206-.018-.372-.17-.52-.084-.087-.99-.715-1.853-1.334-.827-.594-1.623-.965-1.6-1.273.014-.17.114-.647.154-.797.027-.107.324-1.22.432-1.625a.143.143 0 0 1 .183-.098c4.258 1.445 5.568 4.202 8.811 5.912 3.897 2.055 8.205.36 4.885-2.254Z"
            fill={
              {
                "payload": 4291078504,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
        <RNSVGDefs>
          <RNSVGClipPath
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            name="iconprimelogo"
          >
            <RNSVGPath
              d="M0 0h280v61H0z"
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGClipPath>
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        [
          {
            "fontFamily": "ReservaSerif-Regular",
            "fontSize": 20,
          },
          {
            "color": "#ffffff",
            "fontSize": 18,
            "lineHeight": 23,
            "marginBottom": 24,
            "textAlign": "center",
          },
        ]
      }
      variant="subtituloSessoes"
    >
      Um paraíso de benefícios só
      

      seus por 12x
      <Text
        style={
          [
            {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 15,
            },
            {
              "fontSize": 18,
              "fontWeight": "900",
              "lineHeight": 23,
            },
          ]
        }
        variant="descontoTag2"
      >
         25/Mês*.
      </Text>
    </Text>
    <View
      style={
        {
          "paddingHorizontal": 20,
          "width": 576.9230769230769,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "height": 95.28571428571429,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.usereserva:id/PrimeHero_call_to_action"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-Regular",
                "fontSize": 15,
              },
              {
                "fontSize": 12,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "lineHeight": 14,
              },
            ]
          }
          variant="tituloSessao"
        >
          ASSINE AGORA
        </Text>
      </View>
    </View>
  </View>
</View>
`;
